webapp/components/searchable_channel_list.jsx

```diff
@@ -47,6 +47,12 @@ export default class SearchableChannelList extends React.Component {
         }
     }
 
+    componentWillReceiveProps(nextProps) {
+        if (nextProps.isSearch) {
```

Comment:
Doesn't this mean the page will reset any time we're searching and something happens that changes the props? Is there anything we could look at instead that would make this only trigger when the search term changes?

Reply:
Are there any cases where something would change the props mid search? Like when a new channel is created during someone else's search? I could potentially pass in the search term as a prop, and diff it with the current search term.